CMakeLists.txt

```diff
@@ -15,3 +15,5 @@ add_executable(memorypool ${SRC})
 target_link_libraries(memorypool PRIVATE memory_pool)
 
 target_include_directories(memorypool PRIVATE src)
+
+add_dependencies(memorypool update_timestamp)
```

Comment:
The selected line add_dependencies(memorypool update_timestamp) adds a dependency to the memorypool target, specifying that it depends on the `update_timestamp` target. This ensures that the `update_timestamp target` is **built** before the `memorypool` target.